src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -752,5 +753,34 @@ static void WritePropertyAccessorModifiers(CodeWriter writer, MethodSignatureMod
                     .AppendRawIf("private ", modifiers.HasFlag(MethodSignatureModifiers.Private));
             }
         }
+
+        public static void WriteOverloadMethod(this CodeWriter writer, OverloadMethodSignature overloadMethod)
```

Comment:
I would like to suggest we do not need to introduce this method.
Instead, we should build a proper `Method` instance (with a body), and call `writer.WriteMethod(overloadMethod)` where overloadMethod has a type of `Method`
All the calculations could happen when we build the `Method` instance.

Reply:
updated